csrc/id_model/validation_utils.cpp

```diff
@@ -8,106 +8,132 @@
 #include <compute_at_map.h>
 #include <id_model/to_string.h>
 #include <id_model/validation_utils.h>
+#include <ir/utils.h>
+#include <utils.h>
 
 #include <sstream>
 
 namespace nvfuser {
 
-void IdModelValidator::checkExactGraphEquivalence(const ValGraph& exact_graph) {
-  // Empty graph
-  if (exact_graph.disjointValSets().disjointSets().empty()) {
-    return;
-  }
-
-  auto all_exprs = exact_graph.disjointExprSets().getAllElements();
-  if (std::find_if(all_exprs.begin(), all_exprs.end(), [](Expr* expr) {
-        return expr->isA<Swizzle2D>();
-      }) != all_exprs.end()) {
-    // Ignoring a fusion with swizzle
-    return;
+IdModelValidator::IdModelValidator(Fusion* fusion) : ca_map_(fusion) {
+  for (auto tv : ir_utils::allTvs(fusion)) {
+    for (auto id : ir_utils::allIDsOf(tv)) {
+      if (id->definition() && id->definition()->isA<Swizzle2D>()) {
+        has_swizzle_ = true;
+        break;
+      }
+    }
   }
+}
 
-  Fusion* fusion = exact_graph.disjointValSets()
-                       .disjointSets()
-                       .at(0)
-                       ->vector()
-                       .at(0)
-                       ->fusion();
-  ComputeAtMap ca_map(fusion);
-
-  DisjointSets<IterDomain*>& ca_map_exact_sets = ca_map.id_graph_.exact_nodes_;
-
-  // Propgate mappings through expressions in ComputeAtMap. Since we
-  // want to traverse and update ca_map_exact_sets, once  updated, the
-  // traversal of the ID groups cannot continue and needs to be
-  // restarted. The algorithm seems terriblly inefficient, but
-  // shuldn't matter as this is just for transitory validations
-  bool updated = true;
-  while (updated) {
-    updated = false;
-    for (const auto& set : ca_map_exact_sets.disjointSets()) {
-      auto uses = ca_map.uniqueExactUses(set->vector().front());
-      auto use_count = uses.size();
-      // Note that it should be fine to continue updating the map with
-      // the loop below as it should only modify output domain groups
-      for (size_t i = 0; i < use_count; ++i) {
-        auto use_i = uses.at(i);
-        for (size_t j = i + 1; j < use_count; ++j) {
-          auto use_j = uses.at(j);
-          if (!IterDomainGraph::exprsMap(
-                  use_i, use_j, true, ca_map_exact_sets)) {
-            continue;
+void IdModelValidator::fullyPropagateMappings(
```

Comment:
The previous method was fine for the exact map but not for the almost exact map. Changed to more straightforward and naive approach. While the algorithm is really inefficient, it shouldn't matter as this is just for temporary validation.